include/boost/beast/http/impl/basic_parser.hpp

```diff
@@ -56,6 +56,18 @@ put(ConstBufferSequence const& buffers,
         buf_.get(), size}, ec);
 }
 
+template<bool isRequest>
+boost::optional<std::uint64_t>
+basic_parser<isRequest>::
+impl_content_length() const
+{
+    // don't pessimise possible NRVO
+    boost::optional<std::uint64_t> result;
```

Comment:
What is this nonsense of NRVO? `boost::optional<std::uint64_t>` is a trivial type...